jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```
@@ -927,16 +957,150 @@ static String formatExpires(Instant expires)
+                    throw new IllegalStateException("Too many delimiters for a Date format");
+                }
+
+                if (token.isBlank())
+                    continue; // skip blank tokens
+
+                // RFC 6265 - Section 5.1.1 - Step 2.1 - time (00:00:00)
+                // if (hour == (-1) && Immutable.PATTERN_TIME.matcher(token).matches())
+                if (hour == (-1) && token.length() == 8 && token.charAt(2) == ':' && token.charAt(5) == ':')
+                {
+                    hour = StringUtil.toInt(token, 0);
+                    minute = StringUtil.toInt(token, 3);
+                    second = StringUtil.toInt(token, 6);
+                    continue;
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.2
+                if (day == (-1) && token.length() <= 2)
+                {
+                    day = StringUtil.toInt(token, 0);
+                    continue;
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.3
+                if (month == (-1) && token.length() == 3)
+                {
+                    Integer m = Immutable.MONTH_CACHE.getBest(token);
+                    if (m != null)
+                    {
+                        month = m;
+                        continue;
+                    }
+                }
+
+                // RFC 6265 - Section 5.1.1 - Step 2.4
+                if (year == (-1))
+                {
+                    if (token.length() <= 2)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    else if (token.length() == 4)
+                    {
+                        year = StringUtil.toInt(token, 0);
+                    }
+                    continue;
+                }
+            }
+        }
+        catch (Throwable t)
+        {
+            throw new DateTimeSyntaxException("Unable to parse date: " + date, t);
```

I still think technically this is just an ignore and continue.... but fail early is probably better.
Does it still pass tests with an ignore and continue?

---

done